jekyll/_cci2/docker-hub-pull-through-mirror.adoc

```diff
@@ -20,60 +20,135 @@ Starting https://www.docker.com/blog/scaling-docker-to-serve-millions-more-devel
+----
+sudo docker run \
+  -d \
+  -p 443:5000 \
+  --restart=always \
+  --name=through-cache-secure \
+  -v /root/tls:/data/tls \
+  -e REGISTRY_PROXY_REMOTEURL="https://registry-1.docker.io" \
+  -e REGISTRY_PROXY_USERNAME=DOCKER_HUB_USERNAME \
+  -e REGISTRY_PROXY_PASSWORD=DOCKER_HUB_ACCESS_TOKEN \
+  -e REGISTRY_HTTP_TLS_CERTIFICATE=/data/tls/fullchain.pem \
+  -e REGISTRY_HTTP_TLS_KEY=/data/tls/privkey.pem \
+  registry
+----
++
+NOTE:
++
+.. Replace `DOCKER_HUB_USERNAME` and `DOCKER_HUB_ACCESS_TOKEN` with the username for the Docker Hub account and the access token you obtained through https://hub.docker.com/settings/security, respectively.
+.. Replace `fullchain.pem` and `privkey.pem` with the filenames of your certificate and private key, respectively. *The certificate specified there must be correctly chained enough for tracing the certification path from the leaf to the root*.
+
+. Finally, *make sure that the TCP port 443 (i.e. HTTPS) is open and accessible*. For better security, we recommend you to open the port only to Nomad clients and VMs for `machine` executors and remote docker engines.
++
+On AWS, for example, you will need to make sure that both Security Groups and the firewall at the OS level are configured correctly to accept connections from Nomad clients and VMs for `machine`/`setup_remote_docker` jobs over HTTPS.
+
+==== Plan for renewal of TLS certificates
+
+You will need to renew TLS certificates periodically. These are the steps required to renew certificates.
+
+. Update the certificate and the private key in `/root/tls`.
+
+. Run `docker restart through-cache-secure`.
+
+Technically, this can be automated. For example, if you are using Let's Encrypt for your certificates, you can setup a cron task that executes `certbot renew` and the steps above.
+
+== Configure Nomad clients to use the pull through cache registry (run this for _each_ Nomad client)
+
+. Run the command below to *configure the `registry-mirrors` option for the Docker daemon*.
++
+NOTE: Replace `http://192.0.2.1.or.https.your-mirror.example.com` with the URL of your pull through cache registry accordingly.
++
+[source,bash]
+----
+sudo bash -c 'cat <<< $(jq ".\"registry-mirrors\" = [\"http://192.0.2.1.or.https.your-mirror.example.com\"]" /etc/docker/daemon.json) > /etc/docker/daemon.json'
+----
+
+. *Reload Docker daemon* to apply the configuration.
 +
 `sudo systemctl restart docker.service`
 
-Now you should be able to see that Docker images from Docker Hub are downloaded through the cache server. As a side effect, you should be able to see that private images for the user, configured for the cache server can be downloaded without explicit authentications.
+Now you should be able to see that Docker images from Docker Hub are downloaded through the cache registry. As a side effect, you should be able to see that private images for the user, which is configured for the cache registry, can be downloaded without explicit authentications.
```

Comment:
The meaning of the side effect here isn't quite clear to me... what is meant by "the user"?

Reply:
I am referring situations like this:
Assume that `DOCKER_HUB_USERNAME` in the "Set up a pull through cache registry" section is `makotom`.
Without the proxy, a private Docker image `makotom/my-private-image` cannot be used for `docker` executors unless the `auth` block in the config.yml is configured correctly.
Then, once the proxy is configured correctly, `makotom/my-private-image` becomes available for `docker` executors, even without specifying `auth` block in the config.
In this example "the user" in the original sentence equals `makotom`.

Your suggestion on better wording is super appreciated 😹